src/sonic-yang-models/yang-models/sonic-mirror-session.yang

```diff
@@ -99,7 +99,7 @@ module sonic-mirror-session {
                 leaf gre_type {
                     when "current()/../type = 'ERSPAN'";
                     type string {
-                        pattern "0[xX][0-9a-fA-F]*";
+                        pattern "0[xX][0-9a-fA-F]*|[1-9][0-9]*";
```

Comment:
I don't think this should be the pattern. This would allow values out of range. The pattern should restrict values from 0-65535.

Reply:
Good catch. I updated the pattern for matching decimal, and some test cases as well. Not we only accept values in range [0, 65535].
Please be noticed that values as `0x0` and `0` are also accepted.